### PR TITLE
[Bug] Fix text alignment and enable href in Button

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description

I noticed that the text inside the button was misaligned in some of our demos and our 404 page. This was the case for all buttons that were used as anchor tags, so where `href` was used. 

This has been fixed in the vizro-bootstrap repository here: https://github.com/McK-Private/vizro-bootstrap/pull/36

While fixing this I was actually wondering why we did not enable the `href` argument for the `Button`. For me, it functions similarly as our `Card`, where we have also enabled `text` and `href`. This also seems to be a common use-case, so in this PR I would like to add the argument in unless we had good reasons to not do so?

## Screenshot

**Before**

Screenshot 2024-11-19 at 09 37 12

**After**

Screenshot 2024-11-19 at 09 37 19

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/src/vizro/models/_components/button.py
Comment (antonymilne) [inline vizro-core/src/vizro/models/_components/button.py:40]: I was surprised to see that we set `target="_top"` here (as in `vm.Card`) and had to look back at https://github.com/mckinsey/vizro/pull/585 to remind myself why.

Please could you just check this works as expected with _relative_ links though? If you have a dashboard in pycafe with a button and card pointing to another page in the same dashboard, it should not break out of the iframe but instead stay in it. It's only external links that should break out of the iframe. I think this probably works correctly already but would be good to check. If it's not working already then we should only set `target="_top"` conditionally for external links only.

File: vizro-core/changelog.d/20241118_113651_huong_li_nguyen_button_alignment.md
Comment (antonymilne) [inline vizro-core/changelog.d/20241118_113651_huong_li_nguyen_button_alignment.md:22]:
```suggestion
- Enable `href` inside `vm.Button`. ([#881](https://github.com/mckinsey/vizro/pull/881))
```

Dedicated to @petar-qb.

**Review by antonymilne — APPROVED** [review]
Looks good! I had noticed the bad link alignment on buttons when I added one to ViViVo. What is the unminified change to CSS to fix it? Can my custom CSS on 
lines 31-35 here be removed now? https://github.com/mckinsey/vizro/pull/824/files/a304db5e8797bdb53e6fcb53f9cd78fa3d94e3fb#r1815769534

The `Button.href` is probably worth a brief mention explicitly in the docs and crosslink/mention between that `Card.href`.

**Comment by huong-li-nguyen** [comment]
> Looks good! I had noticed the bad link alignment on buttons when I added one to ViViVo. ~What is the unminified change to CSS to fix it?~ Oh, I see you linked to it 👍 ~Can my custom CSS on lines 31-35 here be removed now?~ https://github.com/mckinsey/vizro/pull/824/files/a304db5e8797bdb53e6fcb53f9cd78fa3d94e3fb#r1815769534 Judging by the [pycafe preview](https://py.cafe/snippet/vizro/v1?c=H4sIAJpQPGcC%2F9WdWZPbNhLHvwpWeYiUjKg57J14qrRVju0cVY7tco4XaYoFkZCEFXgsCEoz65rvvt3gBUo0xUntA6DEniGJ4%2FcHuhsACcpfRkESstEdGS31f6%2FTlARJvOabXFLFk5isE0lCmm1XCZWhV6Raxst4LZOIqMeUxxvCozSRivwZQwa8Vh5jNn%2BVJCpTkqZ%2BkMDpmMUqIzQj4SqoE%2B75f2XiRUAi9LV9VJYfbKlU%2FkYmeZpVtbx%2B%2F95%2F88vrz3%2F4P3%2F%2B%2BOenC2Ic%2FA5HmOVnzHFBfo2xTsEU%2B0Q3rCozz1QSmTRlwT8J9vAmiRXlMZMX5Dcqd2FyqKSimCrlhimfZhn8nUtxQbYqEmUqraRK9hceLHWLhGxNIrpjfqEogLYcp8B0VzTaYh95iHiMfD8h039Bc3hvIMPdMibwKbrgNygM2opgSeTA1ZZke%2BgI6LsLIni8YyFRCVFbRiTfbBXByghflz8zUtXiIR0W%2B1pusrIG%2FBRwOjWUg%2BhFVWAOVY7PTOUyzu7%2BfwyVdeHv35A%2FtkyybzMCvUR%2B%2F%2BtnkjJZGARRXAloqwwLp4owb%2BPpQsGqONgaX%2BVoulP4oZhci0dC45CEbM9p%2B%2FwFOWx5sCWrRG2rWrd0z4oKAOfHKuVydAG4UrIMjEbLqgvxQLRX5W66rwTSxeGvXkqhgeZaI5IfkvhbRVbMaCnUW5YEZfoxjRhk0Jk1kCeSA5PjiSdZKmjAxlASgi1H0%2BVoUmSUulMqkxk3HarYg5qvzRYuP%2F9Y3I%2B1MWczHiHzTDdyNvtSQTyhxG%2F02Sn27qTqr%2FLzDXzIl4bzqXW1qPKiObeVbD2vm6S8MDnyk20SMUwDflp65NgIBneGm9cuUiXs9pPqamGoaLKZjm2MggFUnRZqS%2BFNJxb2xmMzEn3NZVp8%2BgdJ1sXZrHGimgMq905dqR%2Bzj6btPJ%2B09VHJwNYkCoNcunPAoahIt3TFFA%2BoIIkMobbCQcI8FXASkz%2FLt6pKz7jYYN%2BiJINhRRQhxzsx7NoiDOvGQudms6DhGkYn6GOSqzlkf69%2FG28kD%2BeLxeUFuYL%2F7%2B%2BRQfgbms6XoxeX6QN4lJG9GS3mi5Z118PEWDuYScBjJRMfT0PZAlwMkaB0fX6K59t14Kc1Bo3b1%2Fo5qk%2FXKDPpTop2VcTiuLSScXdC%2FPiSRcme%2BbWNZKZDeoWJfE%2FaDVD5ka8vH4s1Pzv2OBc0WoXUGHqKcPKVXB2i7o%2BSmhXenwaaDkma8w5icqYW%2FSNzMTQPSXhXucfbxsFKl9IuVngFjJIZBvwi0pM%2F0QPA6PdMZuCOEaNxmRxTFuMyHsEwumOp%2FokX1lxmCkfMqk50ezjfzOBgWIW0By4EDj54DcfXumQ9lidFpioG6yEKQ1jbFVH72Aj75XTBtKqKv2jXyZO3pyKHZp4ch%2FuY7rGSczH%2BA92%2Fh2SdEb4sogicOEzpoFahEyR5JK3RWSf9G7HdOKjjelN%2FZ1w3Lv7N4N0RtEt8Y2ioCm0HzLLVxmY0XDHRFy51f82%2FHKe4Iwvd3zzsCh4n8eCiz60n909Ghdhh7eiJ08iWy9bmOCelr5WKiuC%2FHH2ELt5zdqhnG93REjL%2FQVfZWMFf88WA2cbkuNtQ8%2BJkHfKduRCB%2BFBCYODRCmBG%2B%2FHtRz3GZ1z335aBm8mTDiTZNslFWAQGDpMHGF6gypDveZhD71e0F1imHqAxiiiV3s1mIgH72CaZuvvh8uXlrB6RZ6XjB7rkf8MaCAwkZZXXg2xoJpyv4CG0DCnWd2YUQHc6YHas9pDIHYl5wGAGoD1OJ%2BVMUhlstY2CyfONrhsMso4%2FRee9rQ7HlckftaZXzZ5h2tFMSkqj0pL1hOJ5032vbS5%2F8Qxbcw8ttsoFlY%2B13RTWv6jbmXzXiVcNKY3SeeFt5ZHhcJDEz5hggUpkmehHejzAc8WiznHd8OHCc1u2fowL3gmzGe1Qy9EvcPJ0mnHfPvyeLL4WhTttv2XoTVHliDypDJ6mKXS3XgXDomWVcxGOa0OY6OseHnvF5MwDwxGhZLEHF1hc2Ua4Cjoi2AIDf1iEcL0IwkW492u0GWcymLcW6LBMEskmwUUMtgSBed9ytKIxeDlmpEJpY8CFO6bTaZYjvZJfju6PFy7LEXpaBq62garzlQdBZhYFOx5n7HFW5jIyKSqBBrL5K0HjXeuanhV%2BKGaFNA62iZzWAahOONGtmUF3Q6iYk7rRijNFaMSTMo%2FHeNthDo5%2FBTlA2kiy%2F%2BQcZjkYAvEmT4WePnoBXbPZZjuDjuVrGqgjDTM4BSaczaDYbHZ19ery1c319avLq1nK0yLJ9NK78q5%2F8MDTLqfp4800hlg7pfGjd9gKYMpVshZgVvP5tXfjXS1jaIBgB0cvvBfe9TLmOGzM5y%2B9qxs8TEWixCMeQ4IrpF9zwRB78WWEgw8KeP%2Frm3cffn%2FnqQeFKaB3TVmSHryiV3JonTKodXTQ7Pb29tX1i3%2B%2BoDe3l6tXl%2FTm5Q38dbW6osHL6%2Bub4HZ9G9xerV%2BUQmF1wmbsgeLcIYNzGDemTdyYmVRPF6TB%2Ffzu9dvf3nlRaBFsw9RCRRNKHy3iLIHakMV9CrwE3mwTbBusCzrI4I%2B%2B9%2BjBr%2FaxH%2FF1SVjD0AZcODezj9%2BE64Jv39yCaYqdJtSNOUQQo24oKjnPS1pR6YKiCnOQIHzc4oioGnWAMMayA5WRE8pM1gHSkgecmTihzEAdICxfrQSbRjR1QlubdrA8xSMGaxrmkMYW8lCh7ugbLEsI5obbNaRDZEWO%2BFs03NOMJ6H26zJhz4vDRXnogrAGdJioBO%2BDb11RZtAOkJeIPIqnNA6nroT%2BLuShQt3RN1hWHLNAsXCaBRTd1Q2FXdBDxErJRHGnOqJK8gc31HZSD5CbRznm27NpkMu9G87ZxXxeaggZ5IbHm6kjC9lT4OeIzBQNdmD%2Bzok9Bh8gOolzJ6amDegQUcqVxaCJOkBYHq1WTAgnlJms56WtqROjf4U5QJBIDq5YYYt1gLQcJghOGKFBel7WhsbKiVjYgA4QJXnoyl2%2FFut5aVtGlSMOZqIOEMYzlWwkdeImdBv2vDhX1rHPWLyKRAieJk5YYov1vLSISh6x3S5xQVsb9ry4mCnc%2FOWCNBP1vDC9qdKdBcwx7jMEOvPEoIN4uEx37o51EJ%2BXmVLukLUe0Q6W504nngIPESmpEExALrABHuM7CG5o%2FQr3AMk8cGea1oYdIo65IWtoLJU0dCO%2BNKCDRHEqHFFVkZ6XldF4x5x4CGuQDpBVPE5x6OFIB%2FFgmQ7pGypMJKkTUbEBHSAKRsCdK4vyNuwAcW49yXn%2B85sqhzuTyw7iITJZmjJ3NkGc8A6QmMtU5Nk0ZGsecDVdc5iWOqT4DP75BlCSMUdu55qo54XtWeyEZ9acAyTxBHrWCVEN6XlZB4o76qgbD1jbsF3iqncF7VNSk7Wwze9IsusNrmOyNnb5NUy6WyzjPkbrBK%2FfLbcTvoXXEqBfF575Po%2B58n276E%2FZutDxfdVEcpZZCd%2Bi68LXP%2FxccWGngCO%2BDgnG9j8bFRzhdQiovx3ARvwWXBe88a0HVvIf8XVIqMuwOQx1QvaJ0e822iykAuwTgctomzWUfL0SyhfhrJbRMPZKKZ58Wq2kRjwvxMcFoP1iKsx%2BQdWbRHbrMSj75Bhv2disp43ZK0jfKvNpHPr6forVqk5Z%2B6WV75r41a14u8V10PbJq7fm%2B7aPRSekw2SVd3MdkndE3CtTv3NgtayKsF%2BGsn6wMhh7pVQvFVitxYDsE1NuTrdZSoPYJ6TYj26zjpqwT0a1SdtmIQZjr5R6U7bVYkzKPjnWz3qGTHUiuoGldx7iV3PqR582C%2BqC7RNX7vu0fh5wxDlEkgO9dYraJ6zY3ml9V7UxBwhyoKNOSPtlFVtTfXNrqt3quoF7RXK7I3vJ1yeh2LBps4iasE9GuZXRZh0NYq8QB%2B4kDLx%2FUCbzy%2F2XDigyUHuFOXLf4Bl3C6qkDgxDp6j9wvRmNvvvPR6D9onCr6ZzoauOOHsllbvSrJbTMPZJKXdt2aykQewT0mzSsllLi7JDDn4%2FhI0CKq4O5HoBayN3C64DPsXdZirxD9vEziemJ4AdIiRM1ni8sRHfQOsAz1Kq8F0lC8ENtA5wHCtspK64WsjmP%2Bjg8dgi6GOyr2Pb9W83nKA93T%2F9Dzi5CcwWdQAA) I think the `line-height: unset` can be removed but the `margin-bottom: 12px` is still needed?

Ahh, I didn't realise you had added custom CSS. All of this can be removed now - I'll quickly push that change 👍

File: vizro-core/src/vizro/models/_components/button.py
Comment (huong-li-nguyen) [reply vizro-core/src/vizro/models/_components/button.py:40]: Yes, that seems to work as intended 👍 https://py.cafe/huong-li-nguyen/vizro-button-link-examples

**Comment by antonymilne** [comment]
> Ahh, I didn't realise you had added custom CSS. All of this can be removed now - I'll quickly push that change 👍

I think the `margin-bottom: 12px` will still be needed or the button gets too squashed onto the code snippet.

**Review by petar-qb — APPROVED** [review]
This really looks good.

Thanks for the https://py.cafe/huong-li-nguyen/vizro-button-link-examples that shows how relative and absolute href paths work properly!

**Comment by huong-li-nguyen** [comment]
Will wait for @stichbury approval on the docs changes before merge

File: vizro-core/docs/pages/user-guides/card-button.md
Comment (huong-li-nguyen) [inline vizro-core/docs/pages/user-guides/card-button.md:483]: For review for @stichbury

**Comment by huong-li-nguyen** [comment]
> > Ahh, I didn't realise you had added custom CSS. All of this can be removed now - I'll quickly push that change 👍
> 
> I think the `margin-bottom: 12px` will still be needed or the button gets too squashed onto the code snippet. But yeah, anything to do with positioning of text in the button can hopefully go.

Ahh, i see - it's not related to the the button text

File: vizro-core/docs/pages/user-guides/card-button.md
Comment (stichbury) [inline vizro-core/docs/pages/user-guides/card-button.md:488]:
```suggestion
For a button-style link navigation component, see the [separate guide on creating a link button](#create-a-link-button).
```

File: vizro-core/docs/pages/user-guides/card-button.md
Comment (stichbury) [inline vizro-core/docs/pages/user-guides/card-button.md:489]:
```suggestion
To configure the navigation panel on the left hand side of the screen, refer to the [separate guide on navigation](navigation.md).
```

File: vizro-core/docs/pages/user-guides/card-button.md
Comment (stichbury) [inline vizro-core/docs/pages/user-guides/card-button.md:636]:
```suggestion
To navigate to different pages using a button with an anchor tag, assign an absolute or relative URL to the
```

File: vizro-core/docs/pages/user-guides/card-button.md
Comment (stichbury) [inline vizro-core/docs/pages/user-guides/card-button.md:647]:
```suggestion
Use the `Button.actions` argument to specify which action function executes when the button is clicked.
```

File: vizro-core/docs/pages/user-guides/card-button.md
Comment (stichbury) [inline vizro-core/docs/pages/user-guides/card-button.md:649]:
```suggestion
The example below demonstrates how to configure a button to export the filtered data of a target chart using the
```

**Review by stichbury — APPROVED** [review]
LGTM 🌟 with just a few minor suggestions for clarity.

**Pull request merged**
